test/core/utils/is-context.js

```diff
@@ -0,0 +1,149 @@
+      assert.isTrue(isContextProp({ fromShadowDom: true }));
+    });
+
+    it('is false for other objects', () => {
+      assert.isFalse(isContextProp({}));
+      assert.isFalse(isContextProp({ exclude: [] }));
+      assert.isFalse(isContextProp({ include: true }));
+      assert.isFalse(isContextProp({ runOnly: 'rules' }));
+    });
+
+    it('is false for other types', () => {
+      assert.isFalse(isContextProp(1));
+      assert.isFalse(isContextProp(null));
+      assert.isFalse(isContextProp(undefined));
+    });
+  });
+
+  describe('isContextObject', () => {
+    it('is false if not an object `include` or `exclude`', () => {
+      assert.isFalse(isContextObject(true));
+      assert.isFalse(isContextObject(null));
+      assert.isFalse(isContextObject(1));
+      assert.isFalse(isContextObject({ foo: 'bar' }));
+    });
+
+    it('is true for an object with `include` with a context prop', () => {
+      assert.isTrue(isContextObject({ include: 'string' }));
+      assert.isTrue(
+        isContextObject({ include: document.createElement('div') })
+      );
+      assert.isTrue(isContextObject({ include: [] }));
+      assert.isTrue(isContextObject({ include: { length: 1 } }));
+      assert.isTrue(isContextObject({ include: { fromFrames: true } }));
+      assert.isTrue(isContextObject({ include: { fromShadowDom: true } }));
+    });
+
+    it('is false for an object with `include` that is not a context prop', () => {
+      assert.isFalse(isContextObject({ include: false }));
+      assert.isFalse(isContextObject({ include: null }));
+      assert.isFalse(isContextObject({ include: 123 }));
+      assert.isFalse(isContextObject({ include: { something: 'else' } }));
+    });
+
+    it('is true for an object with `exclude` with a context prop', () => {
+      assert.isTrue(isContextObject({ exclude: 'string' }));
+      assert.isTrue(
+        isContextObject({ exclude: document.createElement('div') })
+      );
+      assert.isTrue(isContextObject({ exclude: [] }));
+      assert.isTrue(isContextObject({ exclude: { length: 1 } }));
+      assert.isTrue(isContextObject({ exclude: { fromFrames: true } }));
+      assert.isTrue(isContextObject({ exclude: { fromShadowDom: true } }));
+    });
+
+    it('is false for an object with `exclude` that is not a context prop', () => {
+      assert.isFalse(isContextObject({ exclud: false }));
+      assert.isFalse(isContextObject({ exclud: null }));
+      assert.isFalse(isContextObject({ exclud: 123 }));
+      assert.isFalse(isContextObject({ exclude: { something: 'else' } }));
+    });
+
+    it('is false if `include` is on the prototype', () => {
+      assert.isFalse(isContextObject(Object.create({ include: 'string' })));
+    });
+
+    it('is false if `exclude` is on the prototype', () => {
+      assert.isFalse(isContextObject(Object.create({ exclude: 'string' })));
+    });
+
+    it('is true for an object with both `include` and `exclude`', () => {
```

Comment:
Is it a bug or a feature that you can have a context object with invalid values for either `include` or `exclude` as long as one is valid? For example `{include: [], exclude: 1}`. Testing it looks like it works without issues.

It might be nice to have a test for it, but I don't think it's a blocker to the pr.